File: 121_best_time_to_buy_and_sell_stock/maxProfit.ts
function maxProfit(prices: number[]): number {
  let max = 0;
  let currentMin = prices[0];

  prices.forEach((price) => {
    const profit = price - currentMin;
    max = Math.max(max, profit);
    currentMin = Math.min(currentMin, price);
  });

  return max;
}

File: 122_best_time_to_buy_and_sell_stock_2/maxProfit.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int curr = prices[0];

        for (const int& price : prices) {
            if (price < curr) {
                curr = price;
            } else {
                profit += price - curr;
                curr = price;
            }
        }

        return profit;
    }
};